server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoConnectionService.java

```diff
@@ -835,6 +849,204 @@ public Response validateConnectionName( final @Context HttpHeaders headers,
         }
     }
     
+    /**
+     * Deploy a VDB for the specified workspace connection
+     * @param headers
+     *            the request headers (never <code>null</code>)
+     * @param uriInfo
+     *            the request URI information (never <code>null</code>)
+     * @param connectionName
+     *            the Connection name being deployed (cannot be empty)
+     * @return a JSON document representing the results of the removal
+     * @throws KomodoRestException
+     *             if there is a problem deploying the vdb
+     */
+    @POST
+    @Path( V1Constants.CONNECTION_DEPLOY_VDB_SEGMENT + StringConstants.FORWARD_SLASH + V1Constants.CONNECTION_PLACEHOLDER )
+    @Produces( { MediaType.APPLICATION_JSON } )
+    @ApiOperation( value = "Deploy a VDB for the workspace connection" )
+    @ApiResponses(value = {
+            @ApiResponse(code = 406, message = "Only JSON is returned by this operation"),
+            @ApiResponse(code = 403, message = "An error has occurred.")
+        })
+    public Response deployVdbForConnection( final @Context HttpHeaders headers,
```

Comment:
I can't help feeling that we are mixing and confusing our resources. Up until now:
* KomodoVdbService has dealt with managing vdbs in the repository;
* KomodoMetadataService has dealt with the teiid server and metadata deployment;
* KomodoConnectionService has dealt with manager connections in the repository.

This seems to now be merging all three. I can see the value of this but I would propose separating this into a different service on its own, maybe KomodoCompositeService or KomodoManagementService?

Reply:
That tells me that there is existing functionality in REST API, but it is not modular enough to re-use in certain ways. I suggest we look as to how to re-use or combine multiple calls to achieve same.